### ISSUE TITLE
Mistake in the Pet Clinic model

### ISSUE THREAD

**Issue by lenaRB** [GraphWalker/graphwalker-example#34]
I am using the Pet Clinic example in a lecture for my students and it kept failing because the 

"actions": [
        "numOfPets=0;"
      ]

is set in OwnerInformation rather than in the NewOwner model, so it keeps trying to add visits or edit pets for owners with 0 pets. Moving it to the correct node solves the problem.

Figured this might save some time to anyone else running into this issue with the example.

Best,

Lena

**Comment by KristianKarl** [comment]
Thanks for the report, I'll look into it.

**Comment by KristianKarl** [comment]
I looked into the models, and the reason for having `numOfPets`in the model OwnerInformation, is to make sure that we do not edit or visit a pet, unless there is a pet to edit or visit.
This is done by the guard in the 2 edges `e_EditPet` and `e_AddVisit` in the model OwnerInformation, which checks that `numOfPets > 0`.

The way the test is designed, we only visit the vertices `v_Pet` and `v_NewVisit` unless we created a new pet for an owner. We could however implement the test to check if there are already existing pets for the selected owner, and set `numOfPets` with that value.

The attribute `numOfPets` is only in scope for the model OwnerInformation. To make it global and shared between models, it has to be named `global.numOfPets`.

I can see a problem if we visited the model OwnerInformation once, adding  a pet, exiting the model, entering it again later but using another owner without pets. Then the value of `numOfPets` would be a mismatch. I suspect that this is the problem you have observed. Did you observe this behavior when running the full java test with PetClinic demo app?

Together with the java implementation of the test, the attribute is set here: https://github.com/GraphWalker/graphwalker-example/blob/1792b8481ed53bf8be635046fe0f25cc18e8f82b/java-petclinic/src/main/java/com/company/modelimplementations/OwnerInformationTest.java#L40
